lntest/itest/lnd_test.go

```diff
@@ -81,9 +82,55 @@ func getTestCaseSplitTranche() ([]*testCase, uint, uint) {
 	return allTestCases[trancheOffset:trancheEnd], threadID, trancheOffset
 }
 
+// cleanProcesses will read the OS's running processes and check whether any
+// old test-related process is running. If found, it will kill the process so
+// that we have a clean state to run our itests. We define "old" as processes
+// not created within one minute.
+func cleanProcesses(t *testing.T) {
+	processes, err := process.Processes()
+	require.NoError(t, err, "failed to read running processes")
+
+	// cleanup is a helper closure that will check the given process's
+	// creation time and kill it if it's older than 1 min.
+	cleanup := func(p *process.Process, name string) {
+		ts, err := p.CreateTime()
+		if err != nil {
+			t.Logf("reading creation time got error: %v", err)
+			return
+		}
+
+		created := time.Unix(0, ts*int64(time.Millisecond))
+		if time.Since(created) < 1*time.Minute {
+			t.Logf("skipped freshly created process: %v", created)
+			return
+		}
+
+		t.Logf("found uncleaned process: %s", name)
+		require.NoErrorf(t, p.Kill(), "failed to kill process")
+		t.Logf("process %s killed", name)
+	}
+
+	for _, p := range processes {
+		n, err := p.Name()
+		if err != nil {
+			t.Logf("reading process name got error: %v", err)
+			continue
+		}
+
+		// TODO: rename bitcoind to bitcoind-itest so we can kill it.
+		if n == "lnd-itest" || n == "btcd-itest" {
+			cleanup(p, n)
+		}
+	}
+}
+
 // TestLightningNetworkDaemon performs a series of integration tests amongst a
 // programmatically driven network of lnd nodes.
 func TestLightningNetworkDaemon(t *testing.T) {
+	// Before running any tests, we need to check whether we have uncleaned
+	// itest-related processes running.
+	cleanProcesses(t)
```

Comment:
> Did you run into problems caused by lingering processes a lot? Or why did you add this (just out of curiosity)?

There's one time I check my machine and found quite a few hanging `lnd-itest` processes, so I want to keep it clean. After this commit I constantly find hanging `lnd-itest`. I guess this is more of a local test thing as the github action always spins up new machines?

> And only the check for the one minute prevents this from killing the other processes launched in itest-parallel?

Yeah exactly. I'm still digging the parallelism here, as I found that, in my final local branch, `make itest` did get all tests passed while `make itest-parallel` didn't.

Reply:
I'm a bit worried about the potential side effects of this commit. This will not affect the jobs run inside of GitHub, correct. We always get new VMs there. But locally it could have weird side effects.

I also think that there will only be residual processes if you abort (Ctrl+C) an itest run. Or if one tranche fails and the other tranches are still completing in the background.

What do you think about just adding a new `make` goal that calls `killall lnd-itest`? Then you could run `make itest-clean && make itest-parallel` if you did run a suite before?

Reply:
Hah yeah, forgot about `make`, that's much easier, switched!